src/xrCommon/xr_smart_pointers.h

```diff
@@ -1,21 +1,38 @@
 #pragma once
 
+#include "../xrCore/xrMemory.h"
 #include <memory>
+#include <functional>
 
 template <typename T>
-using xr_unique_ptr = std::unique_ptr<T>;
+struct xr_custom_deleter
+{
+    void operator()(T* ptr) const noexcept
+    {
+        xr_delete(ptr);
+    }
+};
+
+template <typename T>
+using xr_unique_ptr = std::unique_ptr<T, xr_custom_deleter<T>>;
 
+// TODO: use xr_custom_deleter here
 template <typename T>
 using xr_shared_ptr = std::shared_ptr<T>;
 
 template <class T, class... Args>
 inline xr_unique_ptr<T> xr_make_unique(Args&&... args)
 {
-    return std::make_unique<T>(args...);
+    return xr_unique_ptr<T>(xr_new<T>(std::forward<Args>(args)...));
 }
 
 template <class T, class... Args>
 inline xr_shared_ptr<T> xr_make_shared(Args&&... args)
 {
-    return std::make_shared<T>(args...);
+    //return std::make_shared<T>(args...);
+    return xr_shared_ptr<T>(xr_new<T>(std::forward<Args>(args)...), [](T* ptr)
+    {
+        xr_custom_deleter<T> deleter;
+        deleter(ptr);
```

Comment:
Looks strange, why we need to create an object if we can call xr_delete directly. Maybe it will be better if we implement a deleter for xr_shared_ptr :D

Reply:
For `std::shared_ptr` custom deleter is not a part of type, for `std::unique_ptr` it is. So I've updated `xr_make_shared` to use `xr_custom_deleter` "more directly"